db/users.go

```diff
@@ -116,9 +116,8 @@ func (dbc *DatabaseContext) UpdatePrincipal(ctx context.Context, updates *auth.P
 			changed = true
 		}
 
-		collectionAccessChanged := dbc.RequiresCollectionAccessUpdate(ctx, princ, updates.CollectionAccess)
-		if collectionAccessChanged {
-			changed = true
+		if changed, err = dbc.RequiresCollectionAccessUpdate(ctx, princ, updates.CollectionAccess); err != nil {
```

Comment:
the bug here is that changed means something different than collection changed, so you need to keep `collectionAccessChanged` as a different variable than `changed`